Undefined symbol when importing PLN Python bindings

This is probably related to  opencog/atomspace#2809

```
Traceback (most recent call last):
  File "examples/chase.py", line 13, in <module>
    from opencog.pln import *
ImportError: /usr/local/lib/opencog/libatom_types.so: undefined symbol: _ZN7opencog11ClassServer16update_factoriesEv
```

Well, this is pretty much cheating around the fact that `atombase` and `atom_types` generate a lot of circular dependencies - no matter if I try to link `atom_types` to `atombase` as `INTERFACE` or `PRIVATE`, CMake will point out that:
```
[cmake] CMake Error: The inter-target dependency graph contains the following strongly connected component (cycle):
[cmake]   "atom_types" of type SHARED_LIBRARY
[cmake]     depends on "atombase" (weak)
[cmake]     depends on "value" (weak)
[cmake]     depends on "truthvalue" (weak)
[cmake]   "value" of type SHARED_LIBRARY
[cmake]     depends on "atom_types" (weak)
[cmake]     depends on "atombase" (weak)
[cmake]     depends on "truthvalue" (weak)
[cmake]   "atombase" of type SHARED_LIBRARY
[cmake]     depends on "atom_types" (weak)
[cmake]     depends on "value" (weak)
[cmake]     depends on "truthvalue" (weak)
[cmake]     depends on "value" (strong)
[cmake]   "truthvalue" of type SHARED_LIBRARY
[cmake]     depends on "value" (weak)
[cmake]     depends on "atom_types" (weak)
[cmake]     depends on "atombase" (weak)
[cmake] At least one of these targets is not a STATIC_LIBRARY.  Cyclic dependencies are allowed only among static libraries.
[cmake] CMake Generate step failed.  Build files cannot be regenerated correctly.
```

That change of yours from above pretty much bypasses those safeguards and I don’t think this is worth doing - we can live for now with that `from opencog.atomspace import *` workaround for now.

The only actual fix is to actually deal with this interdependency between `atom_types` and `atombase`, you can see from that error message that they always occur together as dependencies in those other components - perhaps it’s time to combine them into one?

I think that merge you’ve incorporated into AtomSpace should be reverted too.

--- a/.gitignore
+++ b/.gitignore
@@ -121,3 +121,5 @@ Makefile
 ### Linux ###
 *~
 *.autosave
+
+.vscode/

--- a/opencog/cython/opencog/CMakeLists.txt
+++ b/opencog/cython/opencog/CMakeLists.txt
@@ -45,6 +45,7 @@ ADD_LIBRARY(pln_cython SHARED
 ADD_DEPENDENCIES(pln_cython pln-types)
 
 TARGET_LINK_LIBRARIES(pln_cython
+	${NO_AS_NEEDED}
 	pln-types
 	${ATOMSPACE_LIBRARIES}
 	${PYTHON_LIBRARIES}

